Fix dependencies for 3.1.0 package

swainn opened the pull request:
<!--
Thank you for your pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->

swainn committed:
Fix dependencies for 3.1.0 package

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

swainn commented:
@conda-forge-admin, please rerender

swainn committed:
Remove pip check

The Django 4 depenency backports.zoneinfo is only needed for Python < 3.9, but causes issues with the test b/c in Python > 3.9 the package is empty/doesn't exist.

swainn committed:
MNT: Re-rendered with conda-build 3.22.0, conda-smithy 3.21.3, and conda-forge-pinning 2022.11.10.22.07.33

swainn committed:
Merge branch 'main' into fix-deps-3.1.0
